File: frentend/src/app/services/quiz.service.ts
import { HttpClient, HttpHandler } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { map } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor( private _http:HttpClient) { 
    this._http=_http;
  }
  public quiz(){
    return this._http.get(`http://localhost:8080/quiz/allQuizzes`);
  }

  public addQuiz(quiz :any){
    return this._http.post(`http://localhost:8080/quiz/`,quiz);
  }
  public deleteQuiz(qid :Number){
     return this._http.delete(`http://localhost:8080/quiz/${qid}`);
  }

  public getQuiz(qid:Number){
    return this._http.get(`http://localhost:8080/quiz/${qid}`);
  }

  public updateQuiz(quiz:any){
    return this._http.put(`http://localhost:8080/quiz/`,quiz);
  }

  public getQuizzesOfCategory(catId ){
     return this._http.get(`http://localhost:8080/quiz/category/${catId}`);
  }
  public getQuizzesActive(){
    return this._http.get(`http://localhost:8080/quiz/active`);
  }

  public getQuizzCategoryActive(cid){
    return this._http.get(`http://localhost:8080/quiz/category/active/${cid}`);
  }
}

File: backend/examun/src/main/java/com/examun/service/QuizService.java

package com.examun.service;

import com.examun.models.Category;
import com.examun.models.Quiz;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.Set;

public interface QuizService {

    public Quiz addQuiz(Quiz quiz);
    public Quiz updateQuiz (Quiz quiz);
    public Set<Quiz> getQuizzes();
    Quiz getQuiz(Long quizId);
    public  void  deleteQuiz(Long quizId);

     List<Quiz> getQuizzesOfCategory(Category category);

      List <Quiz> getActiveQuizzes();
     List<Quiz> getActiveQuizzesOfCategory(Category c);

}
